trolley/recipient_gateway.py

```diff
@@ -45,18 +45,89 @@ def update(self, recipient_id, body):
         trolley.configuration.Configuration.client(
             self.config).delete(endpoint)
         return True
+    
+    def delete_multiple(self, recipient_ids):
+        if recipient_ids is None:
+            raise InvalidFieldException("Recipient ids cannot be None.")
+        
+        endpoint = '/v1/recipients/'
+        trolley.configuration.Configuration.client(
+            self.config).delete(endpoint, recipient_ids)
+        return True
+
+    def retrieve_logs(self, recipient_id, page=1, pageSize=10):
+        if recipient_id is None:
+            raise InvalidFieldException("Recipient id cannot be None.")
+        
+        endpoint = f'/v1/recipients/{recipient_id}/logs?page={page}&pageSize={pageSize}'
+        response = trolley.configuration.Configuration.client(
+            self.config).get(endpoint)
+
+        logs = []
+        count = 0
+        for log in response['recipientLogs']:
+            temp = trolley.Log.factory(log)
+
+            logs.insert(count, namedtuple("Log", temp.keys())(*temp.values()))
+
+            count = count + 1
+        
+            tempmeta = Meta.factory(response['meta'])
+            logs.insert(count,namedtuple("Meta", tempmeta.keys())(*tempmeta.values()))
+
+        return logs
+    
+    def get_all_payments(self, recipient_id):
+        if recipient_id is None:
+            raise InvalidFieldException("Recipient id cannot be None.")
+        endpoint = f'/v1/recipients/{recipient_id}/payments'
+        response = trolley.configuration.Configuration.client(
+            self.config).get(endpoint)
+        
+        payments = []
+        count = 0
+        for payment in response['payments']:
+            temppayment = trolley.Payment.factory(payment)
+            newpayment = namedtuple("Payment", temppayment.keys())(*temppayment.values())
+            payments.insert(count, newpayment)
+            count = count + 1
+        
+            tempmeta = Meta.factory(response['meta'])
+            payments.insert(count,namedtuple("Meta", tempmeta.keys())(*tempmeta.values()))
+
+        return payments
+
+    def get_all_offline_payments(self, recipient_id):
+        if recipient_id is None:
+            raise InvalidFieldException("Recipient id cannot be None.")
+        endpoint = f'/v1/recipients/{recipient_id}/offlinePayments'
+        response = trolley.configuration.Configuration.client(
+            self.config).get(endpoint)
+        
+        offlinePayments = []
+        count = 0
+        for offlinePayment in response['offlinePayments']:
+            temppayment = trolley.Payment.factory(offlinePayment)
+            newpayment = namedtuple("Payment", temppayment.keys())(*temppayment.values())
+            offlinePayments.insert(count, newpayment)
+            count = count + 1
+        
+            tempmeta = Meta.factory(response['meta'])
+            offlinePayments.insert(count,namedtuple("Meta", tempmeta.keys())(*tempmeta.values()))
```

Comment:
in the next loop, when you do offlinePayments.insert(count, newpayment), I am afraid it will overwrite the meta.

Reply:
Thanks!
Fixed.